x11-misc/adriconf Advanced DRI Configurator

mercuriete opened the pull request:
…gure open source graphics drivers

Closes: https://bugs.gentoo.org/697676
Signed-off-by: Abel Garcia Dorta <mercuriete@yahoo.es>

mercuriete commented:
This PR is almost a copy from
https://github.com/gentoo-mirror/poly-c/tree/master/x11-misc/adriconf

and ported to the new repository and fixing some errors.

thesamesam commented:
@mercuriete Please ask @Polynomial-C if he is okay with you adding it (and ideally if he could commit it).

mercuriete commented:
@thesamesam I need to bump it first to 2.4.1.
https://gitlab.freedesktop.org/mesa/adriconf/-/releases/v2.4.1

and then ask to @Polynomial-C

Polynomial-C commented on x11-misc/adriconf/metadata.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="project">
+		<email>desktop-misc@gentoo.org</email>
+		<name>Gentoo Desktop Miscellaneous Project</name>
+	</maintainer>
```
`desktop-misc` project has been disbanded. See https://marc.info/?l=gentoo-dev&m=160642206127820&w=4

mercuriete replied:
hi @Polynomial-C can you take care of a similar PR?
I don't know how to continue.

I don't know how to give you credit in this PR. All this code was copied from your overlay.

thanks :)

ConiKost replied:
I would be willing to steppin as maintainer.

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Sorry for rotting this PR. Could you update to EAPI=8?

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+inherit cmake desktop
+
+DESCRIPTION="Advanced DRI Configurator"
+HOMEPAGE="https://github.com/jlHertel/adriconf"
```
This seems dead, as it's marked read-only. But you seems to use anyway the newer fork by freedesktop.org?

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.freedesktop.org/mesa/adriconf.git"
+else
+	SRC_URI="https://gitlab.freedesktop.org/mesa/adriconf/-/archive/v${PV}/adriconf-v${PV}.tar.gz"
```
Could you update to 2.4.1?

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+fi
+LICENSE="GPL-3"
+SLOT="0"
+
```
No new line please.

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+src_install() {
+	cmake_src_install
+
+	insinto /usr/share/appdata
```
This is depreated. Please use ```/usr/share/metainfo```.

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
```
Please add the missing inherit ```xdg-utils```.

ConiKost commented on x11-misc/adriconf/adriconf-2.4.ebuild:
```diff
@@ -0,0 +1,81 @@
+	x11-libs/gtk+:3
+	x11-libs/libdrm
+	x11-libs/libX11
+	wayland? ( media-libs/mesa[egl] )
```
This needs to be updated to ```[egl(+)]```, as newest mesa release in tree dropped this use flag and enables this always.

mercuriete commented:
Hey @ConiKost 
You seems to have more knowledge than me.
Can you take over this PR or create a new one?
I will try to address all problems you pointed out but I don't know if I am capable to do It.

Thanks for the feedback, I will try to give It some time on this weekend.

mercuriete commented:
@ConiKost I rebased this PR and fixed the egl(+) problem...
can you review against the new file?
I had to create a new file called 2.4.1 and all your comments now are outdated.

thanks.

mercuriete commented:
I just copied again the files from @Polynomial-C 
https://github.com/gentoo-mirror/poly-c/tree/master/x11-misc/adriconf

Please somebody with more knowledge than me try to push this forward.

juippis commented on x11-misc/adriconf/adriconf-2.4.1.ebuild:
```diff
@@ -0,0 +1,66 @@
+	dev-cpp/glibmm:2
+	dev-cpp/gtkmm:3.0
+	dev-cpp/libxmlpp:3.0
+	dev-libs/boost:=
```
```
CMake Error at /usr/lib64/cmake/Boost-1.77.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_locale"
  (requested version 1.77.0) with any of the following names:

    boost_localeConfig.cmake
    boost_locale-config.cmake

  Add the installation prefix of "boost_locale" to CMAKE_PREFIX_PATH or set
  "boost_locale_DIR" to a directory containing one of the above files.  If
  "boost_locale" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /usr/lib64/cmake/Boost-1.77.0/BoostConfig.cmake:258 (boost_find_component)
  /usr/share/cmake/Modules/FindBoost.cmake:594 (find_package)
  CMakeLists.txt:47 (find_package)


-- Configuring incomplete, errors occurred!
```
This requires `boost[nls]`.

juippis commented on x11-misc/adriconf/adriconf-2.4.1.ebuild:
```diff
@@ -0,0 +1,66 @@
+	KEYWORDS="~amd64 ~x86"
+	S="${WORKDIR}/${MY_P}"
+fi
+LICENSE="GPL-3"
```
```suggestion
LICENSE="GPL-3+"
```
> This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

juippis commented on x11-misc/adriconf/adriconf-2.4.1.ebuild:
```diff
@@ -0,0 +1,66 @@
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="wayland"
```
This USE flag does nothing - did you intend to put some configure option for it?

juippis reviewed (commented):
(Comments apply to -9999 ebuild as well)

Would you be willing to become a proxied maintainer for this, or will you leave it to @conikost 100 %? Since we can't add new packages without maintainers as a general rule.

mercuriete commented:
Hi all I don't have time and knowledge to move this forward.

Thank you for your time to every people involved in this thread.

@ConiKost Could you open a new PR instead of this one? Thank you very much.
